sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -172,13 +172,17 @@ trait CheckAnalysis extends PredicateHelper {
             failAnalysis("Null-aware predicate sub-queries cannot be used in nested " +
               s"conditions: $condition")
 
+          case f @ Filter(condition, _) if PlanHelper.specialExpressionInUnsupportedOperator(f) =>
```

Comment:
why only check filter here? I think we should do it for any operator.

Reply:
@cloud-fan I thought, having a targeted error message for the user would be better. If we do it generically, we would say "Filter" instead of "WHERE clause", right ? Should i have it handled for Filter and have a general catcher as the last case statement to issue a generic error message ?

Reply:
"The operator xxxx contains an unsupported expression type" sounds good

Reply:
Then we can move it closer to the general case?

Reply:
@cloud-fan Sure.